modules/wpa_screenshot_capture/src/Plugin/CaptureUtility/ScreenshotCaptureUtility.php

```diff
@@ -47,17 +84,45 @@ public function capture(array $data = []) {
       $screenCapture->setUserAgentString($data['user_agent']);
     }
     $screenCapture->setDelay($this->configuration['delay']);
+    $filename = $this->getFileName($data);
 
-    // Determine file locations.
-    $entity_id = $data['run_entity']->getConfigEntity()->id();
-    $file_name = preg_replace('/[^a-z0-9]+/', '-', strtolower($data['url']));
-    $file_name .= ".{$this->configuration['image_type']}";
-    $file_path = $this->storagePath($entity_id, $data['run_uuid']) . '/' . $file_name;
-    $real_file_path = \Drupal::service('file_system')->realpath($file_path);
+    // Save screenshot and set our response.
+    $screenCapture->save($filename['real_path']);
+    $this->response = new ScreenshotCaptureResponse($filename['path'], $data['url']);
+
+    return $this;
+  }
+
+  /**
+   * Captures using headless chrome.
+   */
+  public function captureHeadlessChrome(array $data = []) {
+    // Retrieve node/npm path.
+    $config = \Drupal::configFactory();
+    $system_settings = $config->get('web_page_archive.settings')->get('system');
+
+    // Run "npm install" inside module.
+    $wpa_dir = \drupal_get_path('module', 'web_page_archive');
+    $install = "cd {$wpa_dir} && {$system_settings['npm_path']} install";
```

Comment:
Does this `npm install` need to happen prior to every capture? Or can we move it out to be a little less frequent?  It doesn't really hurt anything as of now, but I'd like to evaluate my options here.